Psychtoolbox/PsychOpenGL/PsychGLSLShaders/SmoothedDiscShader.frag.txt

```diff
@@ -48,6 +54,6 @@ void main()
     }
     else {
         gl_FragColor.rgb = baseColor.rgb;
-        gl_FragColor.a = Mod;
+        gl_FragColor.a = baseColor.a*Mod*discAlpha;
```

Comment:
What's the point of having discAlpha, when baseColor.a already exists and has the same effect?

Also note that baseColor.a already encodes the alpha the user can specify as part of the modulateColor argument or alternatively the globalAlpha argument in 'DrawTexture(s)'. Users already know and expect that those alpha parameters will define an alpha value.

Also, once you change the shader to use auxParameter0 for your extra discAlpha, all DrawTextures commands will need to specify that parameter. Iow. all existing user scripts would break in surprising ways as they don't specify discAlpha as auxParameter[0] if this shader would get updated the way you do. Oh and a slight performance impact for passing additional parameters through the pipeline.

So discAlpha is basically a no-go for the existing shader and afaics redundant anyway.

Reply:
OK, I've reverted the addition of the auxParameter in the latest commit, and now use modulateColour to demonstrate the per-disc alpha.